### PR TITLE
#154 TeamsList and ReviewTrustList component api and redux wiring

### LINKED ISSUE #154: Teams list component: Api and redux wiring
Basically what we need to do is wire up the component in #103 and make sure that it actually works.

Create an index.js file if it does not exist. then add this component to the Dashboard page's described in #155. 

At this point you should be able to view the trusts table if you navigate to the dashboard in the frontend

Make a PR at this point just to get a bit of feedback about structuring

Connect the revirew trust list index component to Redux and make it work. You'll need to use a few things correctly:

# mapStateToProps: 

This should be used for fetching the user from the redux store

We fetch all the known users from the store then use the user id to get the right user instance

# mapDispatchToProps:

If the user that we are trying to look at is not in the store, then we'll need to fetch the user from the backend.  

You'll want to make use of:  **apiReduxApps.FETCH_SINGLE_USER.operations.maybeStart**

# useEffect

This is where you'll actyually call the function that dispatches the action that causes the data to be fetched.

### PR DESCRIPTION
**Pull request by babalwa-mbolekwa**
Related issues: [https://github.com/Umuzi-org/Tilde/issues/154]

## Description:

What are you up to? Fill us in :)

## Screenshots/videos
dashboard_teams_and_reviewTrusts

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js [hunk 1]
```diff
@@ -29,21 +29,23 @@ const TeamsTable = (props) => {
...
-                {   
-                    Object.keys(teams).length > 0 ?
-                        Object.values(teams).map((team) => (
-                            <TableRow key={Object.values(teams).indexOf(team)}>
-                                <TableCell>
-                                    {team.name}
-                                </TableCell>
+            {
+                <TableBody>
+                    {
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDetails/TeamsTable/Presentation.js:45]: It looks like you have some extra brackets.

File: frontend/src/components/regions/Dashboard/index.js [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Paper className={classes.paper}>
+            <Typography>TEAMS</Typography>
+            <TeamsTable />
+          </Paper>
+        </Grid>
+        <Grid item xs={12}>
+          <Paper className={classes.paper}>
+            <Typography>REVIEW TRUSTS</Typography>
```
Comment (sheenarbw) [inline frontend/src/components/regions/Dashboard/index.js]: Frontends are about communication. We need to choose words wisely. Dont just give things the same name as the api.

How about "You are a trusted reviewer for these cards:"

File: frontend/src/components/regions/Dashboard/index.js [hunk 3]
```diff
@@ -1,6 +1,37 @@
...
 export default () => {
-  return <Presentation />;
+  const classes = useStyles();
+  
+  return (
+    <React.Fragment>
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <Paper className={classes.paper}>
+            <Typography>TEAMS</Typography>
```
Comment (sheenarbw) [inline frontend/src/components/regions/Dashboard/index.js]: WHY ARE YOU SHOUTING.

You could just use a heading component instead. Look at all the varients you can choose from: https://material-ui.com/components/typography/ 

It's also important to be consistent. Are there any other headings in Tilde that are in all caps? Nope. Not on the board, not on the actions  page, not on the login page...

Who will go update all those other headings to make them all capital letters?  Is that even necessary? Is there a reason why this page should have more capitalisation than the others?

File: frontend/src/components/regions/Dashboard/index.js [hunk 4]
```diff
@@ -1,6 +1,37 @@
 import React from "react";
-import Presentation from "./Presentation";
+import { Grid, Paper, Typography } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import ReviewTrustTable from "../UserDetails/ReviewTrustTable/Presentation";
+import TeamsTable from "../UserDetails/TeamsTable/Presentation";
```
Comment (sheenarbw) [inline frontend/src/components/regions/Dashboard/index.js]: Please make use of the index.js files. If we just draw the PResentation components then none of the redux wiring will actually execute.

Rather:
```
import ReviewTrustTable from "../UserDetails/ReviewTrustTable";
import TeamsTable from "../UserDetails/TeamsTable";
```

If you just import the Presentation then the components wont actually work. They'll just be pictures without functionality. We need the Dashboard to function. We dont just want a bunch of empty tables.

File: frontend/src/components/regions/Dashboard/index.js [hunk 5]
```diff
@@ -1,6 +1,37 @@
...
+            <Typography>TEAMS</Typography>
+            <TeamsTable />
+          </Paper>
+        </Grid>
+        <Grid item xs={12}>
+          <Paper className={classes.paper}>
+            <Typography>REVIEW TRUSTS</Typography>
+            <ReviewTrustTable />
+          </Paper>
+        </Grid>
```
Comment (sheenarbw) [inline frontend/src/components/regions/Dashboard/index.js]: All this layout stuff should be in dashboard/presentation.js. 
Index.js is for smarts, not for layout.

File: frontend/src/components/regions/Dashboard/index.js (hunk 4 above)
Comment (sheenarbw) [reply frontend/src/components/regions/Dashboard/index.js]: Of course these iomports should be in the Dashboard/Presentation.js file. This is not the place for layout

File: frontend/src/components/regions/Dashboard/Presentation.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
 
 export default () => {
-  return <div>TODO</div>;
-};
\ No newline at end of file
+  return <div>TODO</div>
```
Comment (sheenarbw) [inline frontend/src/components/regions/Dashboard/Presentation.js]: All layout should go in here. 
Please import child components and make sure this Presentation file lays them out correctly.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
The file structure needs work.

Also, I see no mention of redux in the code you submitted in this PR.  But the PR title mentions redux. Is the title incorrect or is there code missing?

File: frontend/src/components/regions/UserDetails/TeamsTable/redux/operations.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import Creators from "./actions.js";
+
+export default {
+    ...Creators,
+    openTeamsTable: ({ userId }) =>
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDetails/TeamsTable/redux/operations.js]: What is this for? I dont think the teams table is a modal.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
You included a bunch of weird stuff in your PR. Please remove the unnecessary things.

**Comment by babalwa-mbolekwa** [comment]
teamsDashboard

**Review by Vuyisanani — COMMENTED** [review]
Hi @Babalwa01 

Can you please make sure you have the requested changes resolved, mark them resolve or they should be outdated

**Review by Vuyisanani — COMMENTED** [review]
Don't you also have to write unit tests?

**Comment by babalwa-mbolekwa** [comment]
> You included a bunch of weird stuff in your PR. Please remove the unnecessary things.

attended to changes

File: frontend/src/components/regions/UserDetails/TeamsTable/redux/operations.js (hunk 7 above)
Comment (babalwa-mbolekwa) [reply frontend/src/components/regions/UserDetails/TeamsTable/redux/operations.js]: > What is this for? I dont think the teams table is a modal.

attended to changes requested

**Comment by babalwa-mbolekwa** [comment]
> Don't you also have to write unit tests?

@Vuyisanani not quite sure how I can structure them for this ticket. Kindly assist with the approach

**Review by Vuyisanani — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**